pandas/tests/series/indexing/test_setitem.py

```diff
@@ -309,52 +307,59 @@ class SetitemCastingEquivalents:
         - the setitem does not expand the obj
     """
 
-    @pytest.fixture(params=[np.nan, np.float64("NaN")])
-    def val(self, request):
-        """
-        One python float NaN, one np.float64.  Only np.float64 has a `dtype`
-        attribute.
-        """
-        return request.param
-
-    def check_indexer(self, obj, key, expected, val, indexer):
+    def check_indexer(self, obj, key, expected, val, indexer, is_inplace):
+        orig = obj
         obj = obj.copy()
+        arr = obj._values
+
         indexer(obj)[key] = val
         tm.assert_series_equal(obj, expected)
 
-    def test_int_key(self, obj, key, expected, val, indexer_sli):
+        self._check_inplace(is_inplace, orig, arr, obj)
+
+    def _check_inplace(self, is_inplace, orig, arr, obj):
```

Comment:
maybe can be more generally located?

Reply:
im optimistic we can eventually use this or something like it for a ton of indexing tests, not sure where this particular method/class is going to end up